src/Storages/MergeTree/MutateTask.cpp

```diff
@@ -1896,6 +1900,27 @@ bool MutateTask::execute()
             if (task->executeStep())
                 return true;
 
+            /// Save existing_rows_count before finishing mutation
+            if (ctx->new_data_part->supportLightweightDeleteMutate() && ctx->new_data_part->hasLightweightDelete())
+            {
+                UInt64 existing_rows_count = ctx->source_part->existing_rows_count;
+                if (ctx->has_lightweight_delete)
+                    existing_rows_count = ctx->new_data_part->readExistingRowsCount();
+
+                ctx->new_data_part->existing_rows_count = existing_rows_count;
+
+                /// If existing_rows_count >= rows_count, there must be something wrong while reading it
+                if (existing_rows_count < ctx->new_data_part->rows_count)
+                {
+                    /// Write existing rows count file
+                    auto out_existing_count = ctx->new_data_part->getDataPartStorage().writeFile(
+                        IMergeTreeDataPart::EXISTING_COUNT_FILE_NAME, 4096, ctx->context->getWriteSettings());
+                    writeIntText(existing_rows_count, *out_existing_count);
+                    out_existing_count->finalize();
+                    out_existing_count->sync();
+                }
+            }
+
```

Comment:
I understand it's an easy way to integrate `existing_rows`, but wouldn't it be much nicer and efficient to calculated on delete and update on merge?..